### PR TITLE
Cleanup manipulation of CQ/Cohort resource accounting in tests

### PR DESCRIPTION
**Pull request by vladikkuzn**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #2749

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: hack/internal/tools/go.mod [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module sigs.k8s.io/kueue/internal/tools
 
-go 1.22.4
+go 1.22.6
```
Comment (mbobrovskyi) [inline hack/internal/tools/go.mod:3]: Please don't add this changes, we have separate PR for this https://github.com/kubernetes-sigs/kueue/pull/2859.

**Comment by gabesaba** [comment]
This was not the part of flavorassigner_test which needed refactoring. It is in particular this that we want to delete: https://github.com/kubernetes-sigs/kueue/blob/c257cc03b0d86c6fac29e851de07a29c02e30ec8/pkg/scheduler/flavorassigner/flavorassigner_test.go#L1961-L1962

We want to avoid overriding these fields manually, instead indirectly set this usage/capacity via ClusterQueues. See
https://github.com/kubernetes-sigs/kueue/issues/2502#issuecomment-2228679747

**Comment by gabesaba** [comment]
Additionally this
https://github.com/kubernetes-sigs/kueue/blob/c257cc03b0d86c6fac29e851de07a29c02e30ec8/pkg/scheduler/flavorassigner/flavorassigner_test.go#L1964

All the usage can be manipulated by calling ClusterQueueSnapshot.AddUsage.

Cohort resources can be added via the main test CQ, and in some cases, a secondary test CQ which lends resources and does nothing else

File: pkg/scheduler/flavorassigner/flavorassigner_test.go [hunk 2]
```diff
@@ -295,28 +286,27 @@ func TestAssignFlavors(t *testing.T) {
...
+							"example.com/gpu":     resource.MustParse("3"),
+						}).
+						Obj(),
+				).
+				WithFlavorResource(resources.FlavorResource{Flavor: "two", Resource: corev1.ResourceCPU}, 3_000).
+				WithFlavorResource(resources.FlavorResource{Flavor: "two", Resource: corev1.ResourceMemory}, 10*utiltesting.Mi).
+				WithFlavorResource(resources.FlavorResource{Flavor: "b_one", Resource: "example.com/gpu"}, 3).
+				Obj(),
 		},
+		// TODO:
```
[the remark this answers is not in the document]
Comment (gabesaba) [reply pkg/scheduler/flavorassigner/flavorassigner_test.go:309]: I looked at the first few, and they look right to me

File: pkg/scheduler/flavorassigner/wrappers.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
+package flavorassigner
```
[the remark this answers is not in the document]
Comment (gabesaba) [reply pkg/scheduler/flavorassigner/wrappers.go:1]: It doesn't appear to me that these new builders help express the test logic more concisely - only in a slightly different way. I think we should only be defining builders if it is cumbersome to express some object, and the builder helps make it easier (ResourceGroup is an excellent example).

If you think there's a good argument for going forward with it, let's move it to a separate PR and discuss it there. We want to keep this PR as small as possible so we can focus on the changes related to resource manipulation refactor

**Comment by mszadkow** [comment]
I have a feeling that some test cases names are confusing.
Tests work with one cluster but the test case is "lend", isn't lending between the clusterqueues?
E.g. `"lend try next flavor, found the second flavor"`
wdyt @vladikkuzn @mbobrovskyi @gabesaba ?

File: pkg/scheduler/flavorassigner/wrappers.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbobrovskyi) [reply pkg/scheduler/flavorassigner/wrappers.go:1]: I agree with @gabesaba. Could we split this PR and move this changes to another one?

File: pkg/scheduler/flavorassigner/flavorassigner_test.go [hunk 4]
```diff
@@ -1895,39 +1757,37 @@ func TestAssignFlavors(t *testing.T) {
...
+				Obj(),
 		},
 	}
 	for name, tc := range cases {
 		t.Run(name, func(t *testing.T) {
 			ctx, _ := utiltesting.ContextWithLog(t)
-			if tc.disableLendingLimit {
-				features.SetFeatureGateDuringTest(t, features.LendingLimit, false)
-			}
+			features.SetFeatureGateDuringTest(t, features.LendingLimit, tc.enableLendingLimit)
```
Comment (mbobrovskyi) [inline pkg/scheduler/flavorassigner/flavorassigner_test.go:1790]: LendingLimit feature is enabled by default, so I believe it was correct.

**Comment by mbobrovskyi** [comment]
> I have a feeling that some test cases names are confusing.
> Tests work with one cluster but the test case is "lend",

Or not correct test case.

**Comment by mbobrovskyi** [comment]
> I have a feeling that some test cases names are confusing. Tests work with one cluster but the test case is "lend", isn't lending between the clusterqueues? E.g. `"lend try next flavor, found the second flavor"` wdyt @vladikkuzn @mbobrovskyi @gabesaba ?

Or invalid test case.

File: pkg/scheduler/flavorassigner/flavorassigner_test.go (hunk 4 above)
Comment (mszadkow) [reply pkg/scheduler/flavorassigner/flavorassigner_test.go:1790]: Correct, though this way it's still valid

**Comment by mszadkow** [comment]
> > I have a feeling that some test cases names are confusing. Tests work with one cluster but the test case is "lend", isn't lending between the clusterqueues? E.g. `"lend try next flavor, found the second flavor"` wdyt @vladikkuzn @mbobrovskyi @gabesaba ?
> 
> Or invalid test case.

I will take closer look it's only 4 of them.

File: pkg/scheduler/flavorassigner/flavorassigner_test.go (hunk 4 above)
Comment (mszadkow) [reply pkg/scheduler/flavorassigner/flavorassigner_test.go:1790]: I think I will revert it as there is no effect, no test is setting it to disabled

**Comment by mszadkow** [comment]
@gabesaba @vladikkuzn 
that's another PR that only addresses cohortResource issue: https://github.com/kubernetes-sigs/kueue/pull/3398

**Comment by mszadkow** [comment]
@vladikkuzn you can close this one I believe

**Pull request closed without merging**